tensorflow_quantum/core/src/util_qsim.h

```diff
@@ -308,7 +308,10 @@ tensorflow::Status AccumulateOperators(
   ss.CopyState(source, scratch);
   ss.SetAllZeros(dest);
 
-  for (int i = 0; i < p_sums.size(); i++) {
+  DCHECK_EQ(p_sums.size(), op_coeffs.size());
```

Comment:
Why add the `DCHECK_EQ` ?

Reply:
It's just that there was an existing comment on top of the file that said:
// Assumes p_sums.size() == op_coeffs.size()

It seems to be in line with what the code does, so I figured it would be better to enforce it rather than running the risk of stepping outside an array.

Reply:
Fair enough. Let's keep it then.